include/dlaf/common/range2d.h

```diff
@@ -45,29 +46,113 @@ namespace common {
 
 /// An Iterator returning indices in column-major order.
 template <typename IndexT, class Tag>
-class IteratorRange2D {
-  using index2d_t = Index2D<IndexT, Tag>;
+struct IteratorRange2D {
+  using iterator_category = std::random_access_iterator_tag;
+  using difference_type = SizeType;
+  using value_type = Index2D<IndexT, Tag>;
+  using reference = const Index2D<IndexT, Tag>&;
+  using pointer = const Index2D<IndexT, Tag>*;
```

Comment:
@msimberg

IMHO `IteratorRange2D` should be immutable, since it does not refer to any container and so it cannot change the pointed/referenced value.

AFAIK, this should be legit https://en.cppreference.com/w/cpp/named_req/ForwardIterator, in particular in reference to these lines

>     The type std::iterator_traits<It>::reference must be exactly 
>        - T& if It satisfies LegacyOutputIterator (It is mutable)
>        - const T& otherwise (It is constant)